robot_calibration/include/robot_calibration/ceres/chain3d_to_mesh_error.h

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (C) 2018-2022 Michael Ferguson
+ * Copyright (C) 2015 Fetch Robotics Inc.
+ * Copyright (C) 2013-2014 Unbounded Robotics Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef ROBOT_CALIBRATION_CERES_CHAIN3D_TO_MESH_ERROR_H
+#define ROBOT_CALIBRATION_CERES_CHAIN3D_TO_MESH_ERROR_H
+
+#include <limits>
+#include <string>
+#include <math.h>
+#include <ceres/ceres.h>
+#include <robot_calibration/calibration_offset_parser.h>
+#include <robot_calibration/mesh_loader.h>
+#include <robot_calibration/ceres/calibration_data_helpers.h>
+#include <robot_calibration/models/camera3d.h>
+#include <robot_calibration/models/chain.h>
+#include <robot_calibration_msgs/CalibrationData.h>
+
+namespace robot_calibration
+{
+
+/**
+ * @brief Get the squared distance line segment A-B for point C
+ *
+ * Based on "Real Time Collision Detection", pg 130
+ */
+double distToLine(Eigen::Vector3d& a, Eigen::Vector3d& b, Eigen::Vector3d c)
```

Comment:
can you add comments on what each of the 3 vectors represent? The name of the function makes me expect one of the vectors actually representing the line

Reply:
is "line segment A-B" and "point C" not clear enough?

Reply:
added param documentation

Reply:
I must have totally missed that line segment A-B....somehow... sorry about that!